### PR TITLE
build: upgrade openssl-dev in base-python image

### PR DESCRIPTION
**Pull request by haq204**
## Description
Alpine 3.15 updated openssl to 1.1.1s-r1 so the previous version 1.1.1q-r0 is not available anymore on pkgs.alpinelinux.org for the v3.15 channel. In order to avoid this in the future, we're just going to pin the minor version to 1.1.1 - openssl 1.X does not follow traditional semantic versioning; the scheme is `<major>.<minor>.<minor><patch-letter>`.

Also pin mypy. mypy released a new version [0.990](https://github.com/python/mypy/releases/tag/v0.990) today  which is making some breaking changes causing mypy runs to fail on our existing code. While we should address them, just pinning the mypy version for now so the rules don't change on us unexpectedly

## Related Issues

List related issues.

## Testing

CI

## Checklist

- [ ] **Does my change need to be backported to a previous release?**
  - What backport versions were discussed with the Maintainers in the Issue?

- [ ] **I made sure to update `CHANGELOG.md`.**

   Remember, the CHANGELOG needs to mention:
  - Any new features
  - Any changes to our included version of Envoy
  - Any non-backward-compatible changes
  - Any deprecations

- [ ] **This is unlikely to impact how Ambassador performs at scale.**

   Remember, things that might have an impact at scale include:
  - Any significant changes in memory use that might require adjusting the memory limits
  - Any significant changes in CPU use that might require adjusting the CPU limits
  - Anything that might change how many replicas users should use
  - Changes that impact data-plane latency/scalability

- [ ] **My change is adequately tested.**

   Remember when considering testing:
  - Your change needs to be specifically covered by tests.
    - Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
  - You also need to make sure that the _entire area being changed_ has adequate test coverage.
    - If existing tests don't actually cover the entire area being changed, add tests.
    - This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
  - We should lean on the bulk of code being covered by unit tests, but...
  - ... an end-to-end test should cover the integration points

- [ ] **I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.**

- [ ] **The changes in this PR have been reviewed for security concerns and adherence to security best practices.**

### CODE REVIEW

**Review by LanceEa — DISMISSED** [review]
lgtm...once green. 

Note: we should probably backport this to release/v2.5 branch as well because it will probably be bitten by it too

File: python/requirements-dev.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Dev requirements
 gitpython
 httpretty
-mypy
+mypy<0.990
```
Comment (LanceEa) [inline python/requirements-dev.txt:4]: Is there a reason not to use `=`, if I'm reading the version specifier this says do not allow pre-releases, is that correct?

**Review by LanceEa — APPROVED** [review]
lgtm...let's get this landed to unblock the other PR's. I just had one question to make sure I understood the version specifier syntax.

File: python/requirements-dev.txt (hunk 1 above)
Comment (haq204) [reply python/requirements-dev.txt:4]: it just means use the latest version that's below 0.990. so 0.990 is the upper bound. I just don't know how mypy releases things and it appears they don't do semantic versioning so this is just to cover scenarios if they do a future 0.8x release

**Review by AliceProxy — APPROVED** [review]
gucci gang

**Pull request merged**